parastell/magnet_coils.py

```diff
@@ -14,7 +14,91 @@
+            logger is supplied, a default logger will be instantiated.
+
+    Optional Attributes
+        mat_tag (str): DAGMC material tag to use for magnets in DAGMC
+            neutronics model (defaults to 'magnets').
+    """
+
+    def __init__(self, geom_filename, logger=None, **kwargs):
+        self.logger = logger
+        geom_path = Path(geom_filename).resolve()
+        self.geom_filename = geom_path.name
+        self.export_dir = geom_path.parent
+        self.mat_tag = "magnets"
+        for name in kwargs.keys() & ("mat_tag"):
+            self.__setattr__(name, kwargs[name])
+
+    @property
+    def logger(self):
+        return self._logger
+
+    @logger.setter
+    def logger(self, logger_object):
+        self._logger = log.check_init(logger_object)
+
+    def import_geom_cubit(self):
+        """Import geom file for magnet set into Coreform Cubit."""
+        first_vol_id = 1
+        if cubit_io.initialized:
+            first_vol_id += cubit.get_last_id("volume")
+
+        last_vol_id = cubit_io.import_geom_to_cubit(
+            self.geom_filename, self.export_dir
+        )
+
+        self.volume_ids = list(range(first_vol_id, last_vol_id + 1))
+
+    def mesh_magnets(self, min_size=20.0, max_size=50.0, max_gradient=1.5):
+        """Creates tetrahedral mesh of magnet volumes via Coreform Cubit.
+
+        Arguments:
+            min_size (float): minimum size of mesh elements (defaults to 20.0).
+            max_size (float): maximum size of mesh elements (defaults to 50.0).
+            max_gradient (float): maximum transition in mesh element size
+                (defaults to 1.5).
+        """
+        self._logger.info("Generating tetrahedral mesh of magnet coils...")
+
+        if not hasattr(self, "volume_ids"):
+            self.import_geom_cubit()
+
+        volume_ids_str = " ".join(str(id) for id in self.volume_ids)
+        cubit.cmd(f"volume {volume_ids_str} scheme tetmesh")
+        cubit.cmd(
+            f"volume {volume_ids_str} sizing function type skeleton min_size "
+            f"{min_size} max_size {max_size} max_gradient {max_gradient} "
+            "min_num_layers_3d 1 min_num_layers_2d 1 min_num_layers_1d 1"
+        )
+        cubit.cmd(f"mesh volume {volume_ids_str}")
+
+    def export_mesh(self, mesh_filename="magnet_mesh", export_dir=""):
+        """Creates tetrahedral mesh of magnet volumes and exports H5M format
+        via Coreform Cubit and  MOAB.
+
+        Arguments:
+            mesh_filename (str): name of H5M output file, excluding '.h5m'
+                extension (optional, defaults to 'magnet_mesh').
+            export_dir (str): directory to which to export the H5M output file
+                (optional, defaults to empty string).
+        """
+        self._logger.info("Exporting mesh H5M file for magnet coils...")
+
+        cubit_io.export_mesh_cubit(
+            filename=mesh_filename, export_dir=export_dir
+        )
+
+
+class BuildableMagnetSet(MagnetSet):
```

Comment:
Not convinced by this new name... maybe `FilamentMagnetSet`?

Reply:
Both versions may have filament data associated with them, as it may be desired to build the magnet surface for radial distance finding. `BuildableMagnetSet` is not my favorite either, but I've had a hard time coming up with another distinguishing feature. Maybe CadQueryMagnetSet, or ParastellMagnetSet, or MagnetSetFromFilaments? Could have 3 objects in total as well, `MagnetSet` and `MagnetSetFromFilaments` and `MagnetSetFromGeometry`, which inherit from `MagnetSet`?